pkg/parse/events/funnel_test.go

```diff
@@ -0,0 +1,536 @@
+					Duration: time.Second,
+					Factor:   0, // no backoff, just 1s delay
+					Steps:    3, // at least as many as we expect
+				},
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  Event{Type: RetrySyncEventType},
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType},
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType},
+					Result: Result{},
+				},
+			},
+		},
+		{
+			name: "RetryEvents From RetryBackoff with max steps",
+			builder: &PublishingGroupBuilder{
+				RetryBackoff: wait.Backoff{
+					Duration: time.Second,
+					Factor:   2, // double the delay each time
+					Steps:    3,
+				},
+			},
+			steps: []time.Duration{
+				time.Second,
+				3 * time.Second, //+2
+				7 * time.Second, //+4
+				8 * time.Second, //+1
+			},
+			expectedEvents: []eventResult{
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 1s
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 3s
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 7s
+					Result: Result{},
+				},
+				// No more retry events until another event sets ResetRetryBackoff=true
+			},
+		},
+		{
+			name: "RetryEvents From RetryBackoff with ResetRetryBackoff",
+			builder: &PublishingGroupBuilder{
+				RetryBackoff: wait.Backoff{
+					Duration: time.Second,
+					Factor:   2, // double the delay each time
+					Steps:    3,
+				},
+				SyncPeriod: 10 * time.Second,
+			},
+			steps: []time.Duration{
+				time.Second,
+				3 * time.Second, //+2
+				7 * time.Second, //+4
+				10 * time.Second,
+				11 * time.Second, //+1
+			},
+			expectedEvents: []eventResult{
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 1s
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 3s
+					Result: Result{},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 7s
+					Result: Result{},
+				},
+				{
+					Event: Event{Type: SyncEventType}, // 10s
+					Result: Result{
+						ResetRetryBackoff: true,
+					},
+				},
+				{
+					Event:  Event{Type: RetrySyncEventType}, // 11s
+					Result: Result{},
+				},
+			},
+		},
+		{
+			name: "SyncEvents, StatusEventType, ResyncEvents",
+			builder: &PublishingGroupBuilder{
+				SyncPeriod:             700 * time.Millisecond,
+				SyncWithReimportPeriod: 4000 * time.Millisecond,
+				StatusUpdatePeriod:     300 * time.Millisecond,
+			},
+			// Explicit steps to avoid race conditions that make validation difficult.
+			steps: []time.Duration{
+				300 * time.Millisecond,
+				600 * time.Millisecond,
+				700 * time.Millisecond,
+				1000 * time.Millisecond,
+				1300 * time.Millisecond,
+				1400 * time.Millisecond,
+				1700 * time.Millisecond,
+				2000 * time.Millisecond,
+				2100 * time.Millisecond,
+				2400 * time.Millisecond,
+				2700 * time.Millisecond,
+				2900 * time.Millisecond,
```

Comment:
This should be fired at 2800ms for the SyncEventType.

Reply:
nice catch! 
I fixed that one and all the subsequent ones up to the 4000ms reimport.